onnxruntime/core/providers/cpu/tensor/expand.cc

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "expand.h"
+#include <cmath>
+
+namespace onnxruntime {
+
+#define REG_EXPAND_KERNEL(TYPE)                                                    \
+  ONNX_CPU_OPERATOR_VERSIONED_TYPED_KERNEL(                                        \
+      Expand,                                                                      \
+      8,                                                                           \
+      12,                                                                          \
+      TYPE,                                                                        \
+      KernelDefBuilder().TypeConstraint("T", DataTypeImpl::GetTensorType<TYPE>()), \
+      Expand<TYPE>);                                                               \
+  ONNX_CPU_OPERATOR_TYPED_KERNEL(                                                  \
+      Expand,                                                                      \
+      13,                                                                          \
+      TYPE,                                                                        \
+      KernelDefBuilder().TypeConstraint("T", DataTypeImpl::GetTensorType<TYPE>()), \
+      Expand<TYPE>);
+
+REG_EXPAND_KERNEL(float)
+REG_EXPAND_KERNEL(double)
+REG_EXPAND_KERNEL(int8_t)
+REG_EXPAND_KERNEL(int16_t)
+REG_EXPAND_KERNEL(int32_t)
+REG_EXPAND_KERNEL(int64_t)
+REG_EXPAND_KERNEL(uint8_t)
+REG_EXPAND_KERNEL(uint16_t)
+REG_EXPAND_KERNEL(uint32_t)
+REG_EXPAND_KERNEL(uint64_t)
+REG_EXPAND_KERNEL(bool)
+REG_EXPAND_KERNEL(MLFloat16)
+
+template <typename T>
+Status Expand<T>::Compute(OpKernelContext* context) const {
+  auto input_tensor = context->Input<Tensor>(0);
+  auto input_data = input_tensor->template Data<T>();
+  auto input_shape = input_tensor->Shape().GetDims();
+
+  auto input_dims = input_shape.data();
+  auto input_dims_size = static_cast<int64_t>(input_shape.size());
+
+  auto shape_tensor = context->Input<Tensor>(1);
+  auto shape_dims = shape_tensor->Data<int64_t>();
+  std::vector<int64_t> output_shape{shape_dims, shape_dims + shape_tensor->Shape().Size()};
+
+  if (input_shape.size() > output_shape.size()) {
+    ORT_THROW("Invalid expand shape");
```

Comment:
Return a Status object instead of throwing here. Several other places in this function.